cmd/wallet-sdk-js/jsinterop/async_runner.go

```diff
@@ -0,0 +1,60 @@
+//go:build js && wasm
+
+/*
+Copyright Gen Digital Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package jsinterop
+
+import (
+	"fmt"
+	"sync"
+	"syscall/js"
+)
+
+type AsyncFunc func(this js.Value, args []js.Value) (any, error)
+
+var (
+	jsErr     js.Value = js.Global().Get("Error")
+	jsPromise js.Value = js.Global().Get("Promise")
+)
+
+type AsyncRunner struct {
+	serializeCallsMutex sync.Mutex
+}
+
+func (r *AsyncRunner) CreateAsyncFunc(innerFunc AsyncFunc) js.Func {
+	return js.FuncOf(func(this js.Value, args []js.Value) any {
+		handler := js.FuncOf(func(_ js.Value, promFn []js.Value) any {
+			resolve, reject := promFn[0], promFn[1]
+
+			go func() {
+				defer func() {
+					if r := recover(); r != nil {
+						reject.Invoke(jsErr.New(fmt.Sprint("panic:", r)))
+					}
+				}()
+
+				// All Go functions call invoked in goroutines to avoid blocking the JS side.
```

Comment:
I think you mean `All Go functions are invoked in goroutines to avoid blocking the JS side.`?

Reply:
JS will wait until the Go function is finished. but this can block the browser. So we run the Go function in goruntine and return immediately. After the Go code is finished, it resolves the promise.

Reply:
My comment was about your comment - you say "functions call invoked", which appears to be a typo. I imagine you meant to say either

`All Go functions are invoked in goroutines to avoid blocking the JS side.`
or
`All Go function calls are invoked in goroutines to avoid blocking the JS side.`

Actually, I'm guessing you meant the second one